src/code/cutscene_camera.c

```diff
@@ -353,28 +357,579 @@ void CutsceneCamera_Reset(void) {
     sCurCsCamera->state = CS_CAM_STATE_UPDATE_ALL;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/cutscene_camera/func_80161C20.s")
+// Linear interpolation from initial values to cmd values. Set weight to 100 to go all the way to the cmd value.
+s16 CutsceneCamera_Interp_Lerp(Vec3f* camPos, f32* camFov, s16* camRoll, CsCmdCamPoint* pointCmd, CsCmdCamMisc* miscCmd,
+                               CutsceneCameraInterp* interpState) {
+    f32 lerp;
+
+    if (interpState->type != CS_CAM_INTERP_LERP) {
+        // Initialize
+        interpState->type = CS_CAM_INTERP_LERP;
+        interpState->waypoint = 0;
+        interpState->curFrame = 0;
+        interpState->duration = 1;
+        if (camPos != NULL) {
+            interpState->initPos.x = camPos->x;
+            interpState->initPos.y = camPos->y;
+            interpState->initPos.z = camPos->z;
+        }
+
+        if (camFov != NULL) {
+            interpState->initFov = *camFov;
+        }
+
+        if (camRoll != NULL) {
+            interpState->initRoll = *camRoll;
+        }
+    }
+
+    interpState->curFrame++;
+
+    lerp = ((f32)interpState->curFrame / pointCmd->duration) * (pointCmd->weight / 100.0f);
+
+    if (camPos != NULL) {
+        VEC3F_LERPIMPDST(camPos, &interpState->initPos, &pointCmd->pos, lerp);
+    }
+
+    if (camFov != NULL) {
+        *camFov = LERPIMP(interpState->initFov, miscCmd->fov, lerp);
+    }
+
+    if (camRoll != NULL) {
+        s16 new_var;
+        s16 temp;
+
+        new_var = CAM_DEG_TO_BINANG(miscCmd->roll);
+
+        temp = (s16)(new_var - (s16)interpState->initRoll);
+
+        *camRoll = (s16)interpState->initRoll + (s16)(temp * lerp);
+
+        // *camRoll = LERPIMP((s16)interpState->initRoll, (s16)CAM_DEG_TO_BINANG(miscCmd->roll), lerp);
```

Comment:
I'm wondering why this is here? Is it equivalent code from somewhere else? Is this what we'd like to write, but can't? It's probably worth a comment explaining what this is and why it's here.

Reply:
Ah yeah, this and your next comment were sort of notes trying to work out the logic of the function. I can remove